From 1a87493014050e3bd94000cd36122c3cadf21270 Mon Sep 17 00:00:00 2001
From: Kinga Tanska <kinga.tanska@intel.com>
Date: Tue, 10 Dec 2019 12:21:21 +0100
Subject: [PATCH] Change warning message
Git-commit: 1a87493014050e3bd94000cd36122c3cadf21270
Patch-mainline: mdadm-4.1+
References: jsc#SLE-10078, jsc#SLE-9348

In commit 039b7225e6 ("md: allow creation of mdNNN arrays via
md_mod/parameters/new_array") support for name like mdNNN
was added. Special warning, when kernel is unable to handle
request, was added in commit 7105228e19
("mdadm/mdopen: create new function create_named_array for
writing to new_array"), but it was not adequate enough,
because in this situation mdadm tries to do it in old way.
This commit changes warning to be more relevant when
creating RAID container with "/dev/mdNNN" name and mdadm
back to old approach.

Signed-off-by: Kinga Tanska <kinga.tanska@intel.com>
Signed-off-by: Jes Sorensen <jsorensen@fb.com>
Signed-off-by: Coly Li <colyli@suse.de>

---
 mdopen.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mdopen.c b/mdopen.c
index 98c54e4..245be53 100644
--- a/mdopen.c
+++ b/mdopen.c
@@ -120,7 +120,8 @@ int create_named_array(char *devnm)
 		close(fd);
 	}
 	if (fd < 0 || n != (int)strlen(devnm)) {
-		pr_err("Fail create %s when using %s\n", devnm, new_array_file);
+		pr_err("Fail to create %s when using %s, fallback to creation via node\n",
+			devnm, new_array_file);
 		return 0;
 	}
 
-- 
2.25.0

